src/v2/providers/schedule.ts

```diff
@@ -0,0 +1,182 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2022 Firebase
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import * as express from 'express';
+import { copyIfPresent } from '../../common/encoding';
+import { timezone } from '../../common/timezone';
+import * as logger from '../../logger';
+import { ManifestEndpoint, ManifestRequiredAPI } from '../../runtime/manifest';
+import * as options from '../options';
+import { HttpsFunction } from './https';
+
+/** @hidden */
+interface ScheduleArgs {
+  schedule: string;
+  timeZone?: timezone;
+  retryCount?: number;
+  maxRetrySeconds?: number;
+  minBackoffSeconds?: number;
+  maxBackoffSeconds?: number;
+  maxDoublings?: number;
+  opts: options.GlobalOptions;
+}
+
+/** @internal */
+export function getOpts(args: string | ScheduleOptions): ScheduleArgs {
+  if (typeof args === 'string') {
+    return {
+      schedule: args,
+      opts: {} as options.GlobalOptions,
+    };
+  }
+  return {
+    schedule: args.schedule,
+    timeZone: args.timeZone,
+    retryCount: args.retryCount,
+    maxRetrySeconds: args.maxRetrySeconds,
+    minBackoffSeconds: args.minBackoffSeconds,
+    maxBackoffSeconds: args.maxBackoffSeconds,
+    maxDoublings: args.maxDoublings,
+    opts: args as options.GlobalOptions,
+  };
+}
+
+/** The Cloud Function type for Schedule triggers. */
+export interface ScheduleFunction extends HttpsFunction {
+  __requiredAPIs?: ManifestRequiredAPI[];
+  run(data: express.Request): void | Promise<void>;
```

Comment:
Do we really need to take an express request? Do we really pass that to customers? This makes mocking/testing difficult.

Reply:
Looking at the code here, it looks like we can start by making `run` take no arguments. Maybe we can take a look at the HTTP request that Cloud Scheduler sends to see if there's anything we'd like to expose (e.g. scheduled deliver time and retry count) but we should try to do something smarter than just forwarding a hard-to-mock object.

Reply:
Per our discussion in chat, I've changed this to a `ScheduleEvent` object